pkg/scheduler/util/utils.go

```diff
@@ -115,13 +126,58 @@ func PatchPodStatus(cs kubernetes.Interface, old *v1.Pod, newStatus *v1.PodStatu
 		return nil
 	}
 
-	_, err = cs.CoreV1().Pods(old.Namespace).Patch(context.TODO(), old.Name, types.StrategicMergePatchType, patchBytes, metav1.PatchOptions{}, "status")
-	return err
+	backoff := wait.Backoff{
+		Duration: retryBackoffInitialDuration,
+		Factor:   retryBackoffFactor,
+		Jitter:   retryBackoffJitter,
+		Steps:    retryBackoffSteps,
+	}
+
+	patchFn := func() (bool, error) {
+		_, err = cs.CoreV1().Pods(old.Namespace).Patch(context.Background(), old.Name, types.StrategicMergePatchType, patchBytes, metav1.PatchOptions{}, "status")
+		if err == nil {
+			return true, nil
+		}
+
+		switch typedErr := err.(type) {
```

Comment:
should we have this validation in a function? And then you can use `client-go/util/retry.OnError`.

Reply:
Sure. 👍